### DISCUSSION TITLE
ImageMagick php extension install not working at all.

### DISCUSSION THREAD

**Discussion by Paul760** [ImageMagick/ImageMagick#2256] [Help]
I try to install ImageMagick and iMagick at my webserver with centos-release-7-8.2003.0.el7.centos.x86_64
I want to make .png files from .svg files

I followed the steps from:
https://www.tecmint.com/install-imagemagick-in-linux/

# yum install php-pear php-devel gcc
# yum install ImageMagick ImageMagick-devel ImageMagick-perl

Installing ImageMagick from Repository was no problem and with:
# convert --version
Version: ImageMagick 6.9.10-68 Q16 x86_64 2020-04-01 https://imagemagick.org
Copyright: © 1999-2019 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP(3.1) 
Delegates (built-in): bzlib cairo fontconfig freetype gslib jng jpeg lcms ltdl lzma openexr pangocairo png ps rsvg tiff wmf x xml zlib

Looks like it all works. But to get it running with PHP I did:

# pecl install imagick 
# echo extension=imagick.so >> /etc/php.ini
# service httpd restart

But then
# php -m | grep imagick
gives no output at all.
And my php-info() does not show any ImageMagick at all.

What is going wrong?

After that I tried to install ImageMagick 7 from Source Code
# yum groupinstall 'Development Tools'
# yum -y install bzip2-devel freetype-devel libjpeg-devel libpng-devel libtiff-devel giflib-devel zlib-devel ghostscript-devel djvulibre-devel libwmf-devel jasper-devel libtool-ltdl-devel libX11-devel libXext-devel libXt-devel lcms-devel libxml2-devel librsvg2-devel OpenEXR-devel php-devel
# wget https://www.imagemagick.org/download/ImageMagick.tar.gz
# tar xvzf ImageMagick.tar.gz
# cd ImageMagick*
# ./configure
# make
# make install

But then at # make I get an error:
```
  CCLD     utilities/magick
MagickCore/.libs/libMagickCore-7.Q16HDRI.so: undefined reference to `jpeg_default_qtables'
collect2: error: ld returned 1 exit status
make[1]: *** [utilities/magick] Error 1
make[1]: Leaving directory `/etc/ImageMagick-7.0.10-23'
make: *** [all] Error 2
```

What could I do? Should 6.9 work with the PHP extension. Or should I get version 7 running. Does somebody know how to fix one of the two problems? So I can get the ImageMagick running with the php extension?

**Comment by urban-warrior** [comment]
The `jpeg_default_qtables' method is in your JPEG delegate library.  Its only available if your JPEG headers specify a version greater or equal to 70.  You may have more than one version of JPEG installed and that's causing the problem.
